### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by xplosunn**

### CODE REVIEW

File: src/test/scala/codacy/metrics/PDependSpec.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package codacy.metrics
+
+import codacy.docker.api.Source
+import codacy.docker.api.metrics.{FileMetrics, LineComplexity}
+import com.codacy.api.dtos.Languages
+import org.specs2.mutable.Specification
+
+class PDependSpec extends Specification {
```
Comment (bmbferreira) [inline src/test/scala/codacy/metrics/PDependSpec.scala]: Just to be sure, add a test to check the Failure when a different language is provided.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+* Before sending a pull request make sure your code is tested.
+
+* Before sending a pull request for a feature, be sure to run tests with `./scripts/test.sh`.
+
+* Use the same coding style as the rest of the codebase, most of the checks can be performed with `./scripts/lint.sh`.
+
+* Use `git rebase` (not `git merge`) to sync your work from time to time with the master branch.
+
+* After creating your pull request make sure the build is passing on [CircleCI](https://circleci.com/gh/codacy/codacy-metrics-radon) and that [Codacy](https://www.codacy.com/app/Codacy/codacy-metrics-radon) is also confident in the code quality.
```
Comment (bmbferreira) [inline CONTRIBUTING.md]: radon

File: src/main/scala/codacy/metrics/Tool.scala [hunk 3]
```diff
@@ -0,0 +1,118 @@
+package codacy.metrics
+
+import codacy.docker.api.metrics.LineComplexity
+import com.codacy.docker.api.utils.{CommandResult, CommandRunner}
```
Comment (rtfpessoa) [inline src/main/scala/codacy/metrics/Tool.scala]: Why don't we parse the output immediately for the needed format?

File: src/main/scala/codacy/metrics/PDepend.scala [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        Failure(new Exception(s"Tried to run PDepend metrics with language: $language. Only PHP supported."))
+      case _ =>
+        val filesToAnalyse: Set[File] = files match {
+          case Some(filesSpecified) =>
+            filesSpecified.map(srcFile => File(source.path) / srcFile.path)
+          case None =>
+            File(source.path).listRecursively().filter(_.isRegularFile).toSet
+        }
+
+        runMetrics(source.path, filesToAnalyse).map(_.toList)
```
Comment (pedrocodacy) [inline src/main/scala/codacy/metrics/PDepend.scala]: remove mapping of successful try with toList - instead change runMetrics to return a List explicitly (you can use collection.breakOut on the for-comprehension)

File: src/main/scala/codacy/metrics/PDepend.scala [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    }
+  }
+
+  def runMetrics(directory: String, files: Set[File]): Try[Seq[FileMetrics]] = {
+    for {
+      fileOutputStr <- run(directory)
+      xml <- Try(XML.loadString(fileOutputStr))
```
Comment (pedrocodacy) [inline src/main/scala/codacy/metrics/PDepend.scala]: Use custom configuration for SAXParser used on codacy/codacy-eslint

Comment (pedrocodacy) [reply src/main/scala/codacy/metrics/PDepend.scala]: see https://github.com/codacy/codacy-eslint/blob/native-package/src/main/scala/codacy/XML.scala

File: src/main/scala/codacy/metrics/PDepend.scala [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        runMetrics(source.path, filesToAnalyse)
+    }
+  }
+
+  def runMetrics(directory: String, files: Set[File]): Try[List[FileMetrics]] = {
+    for {
+      fileOutputStr <- run(directory)
+      xml <- Try(XML.loadString(fileOutputStr))
+      metrics <- parseMetrics(xml, directory, files)
+        .toRight(new Exception("Could not parse XML returned from tool"))
```
Comment (rtfpessoa) [inline src/main/scala/codacy/metrics/PDepend.scala]: What was the XML, please log.

**Pull request merged**